include/nuttx/serial/serial.h

```diff
@@ -283,6 +283,22 @@ struct uart_ops_s
                           FAR const void *buf, size_t len);
 };
 
+/* This structure is used for U(S)ART frame, overrun, parity and brk error
+ * counters. This way applications will know if the UART communition is
+ * having some trouble.
+ */
+
+#ifdef CONFIG_SERIAL_TIOCGICOUNT
```

Comment:
why need a new CONFIG_SERIAL_TIOCGICOUNT for a simple new serial_icounter_s

Reply:
My initial idea was to avoid creating a struct when it wasn't necessary, but I agree, this code doesn't need this this new config. The compilation will remove it case not used